taipy/core/pipeline/pipeline.py

```diff
@@ -52,12 +52,13 @@ def __init__(
         pipeline_id: PipelineId = None,
         parent_id: Optional[str] = None,
         subscribers: List[Callable] = None,
+        is_consistent=None
     ):
         self.config_id = _validate_id(config_id)
         self.id: PipelineId = pipeline_id or self._new_id(self.config_id)
         self.parent_id = parent_id
         self._tasks = {task.config_id: task for task in tasks}
-        self.is_consistent = self.__is_consistent()
+        self.is_consistent = is_consistent or self.__is_consistent()
```

Comment:
I don't think your new implementation works regarding `is_consistent`. The is_consistent attribute must be recomputed each time we modify the dag.

Reply:
Is it necessary when we are getting the Pipeline from our repo?  
This function is time-consuming.